Add unit tests to devel env; remove duplicated content from README

jflory7 opened the pull request:
This commit does three things:

* Adds introduction to ListenBrainz unit tests in the _Set up a development environment_ guide
* Remove duplicated content from the README and point to the full development environment guide in the documentation
* Cancels #285 

The second one is more noteworthy. I added an _About_ section to the README to make it more clear to a passerby what the site is. I removed a lot of content that was already in the [development environment instructions](https://listenbrainz.readthedocs.io/en/latest/dev/devel-env.html) to prevent duplicate sets of data (and causing one to inevitably lag behind later).

Later, once more of the contributing-oriented docs are added, I want to add more links to those.

paramsingh commented on README.md:
```diff
@@ -1,103 +1,74 @@
-Where <your_host> is the DNS name or IP address of the machine running ListenBrainz.
-An example callback url if you're running the server on your own local machine would
-be
+ListenBrainz uses [Google BigQuery](https://cloud.google.com/bigquery/ "BigQuery
```
This link doesn't render well in GH markdown? 
![screenshot_2017-11-12_14-21-59](https://user-images.githubusercontent.com/1837631/32697371-ef235794-c7b4-11e7-9277-7c10f26e8628.png)

jflory7 commented:
@paramsingh Oops, nice catch! This happens when I make pull requests at 2am…

I fixed the rendering issue for the BigQuery link. I also fixed the link for the goals page (I originally linked to the roadmap). Should be all set now.

paramsingh commented on README.md:
```diff
@@ -1,103 +1,74 @@
-[Website](https://listenbrainz.org) | [Beta website](https://beta.listenbrainz.org) | [Bug tracker](https://tickets.metabrainz.org/projects/LB/issues)
+[Website](https://listenbrainz.org) |
+[Documentation](https://listenbrainz.readthedocs.io) |
+[Beta website](https://beta.listenbrainz.org) |
```
I think we should remove this link considering it's been down for a long time now.

paramsingh commented on docs/dev/devel-env.rst:
```diff
@@ -95,50 +107,89 @@ Next, open the file with your favorite text editor and look for this section.
+not use volumes that link to data outside of the containers, so it does not
+interfere with production databases.
+
+Also, run the **integrations tests** for ListenBrainz.
```
Sorry for being nitpicky!

jflory7 replied:
@paramsingh Nah, this is why we do code / PR reviews. 😄 The nitpicky details are helpful.

jflory7 commented:
@paramsingh Done! I added a quick fix for your requested changes and rebased my commits down to one.

I prefer pointing people to our docs and not creating a new file. Ideally, we'd mention anything worth highlighting in the `CONTRIBUTING.md` file ( #287 ). I think it makes more sense to be consistent in where we point people and try to make those docs highly visible.

paramsingh commented:
Cool, makes sense!
